packages/sdk-drop-in/src/lib/MoniteIframeAppCommunicator.ts

```diff
@@ -0,0 +1,221 @@
+export class MoniteIframeAppCommunicator {
+  private readonly targetWindow?: Window;
+  private readonly isHost: boolean;
+  private readonly listeners = new Map<
+    string,
+    (payload: ListenerPayload) => void
+  >();
+  private readonly slots = new Map<
+    string,
+    { payload: PayloadSlot; abortController?: AbortController }
+  >();
+
+  private connectionToHostRetryInterval?: number;
+  private slotQueue = new Set<string>();
+  private connected = false;
+
+  static allowedSlots = ['locale', 'theme', 'fetch-token'];
+
+  constructor(iframeElement: HTMLIFrameElement | Window) {
+    if (iframeElement instanceof HTMLIFrameElement) {
+      this.isHost = true;
+      this.targetWindow = iframeElement.contentWindow ?? undefined;
+    } else {
+      this.isHost = false;
+      this.targetWindow = iframeElement;
+    }
+  }
+
+  public connect() {
+    if (typeof window === 'undefined' || typeof document === 'undefined')
+      return void console.warn('Cannot connect in non-browser environment');
+
+    this.disconnect();
+
+    if (!this.targetWindow) throw new Error('Target window is not set');
+
+    window.addEventListener('message', this.onWindowMessage);
+
+    if (!this.isHost) {
+      this.connectionToHostRetryInterval = setInterval(() => {
+        if (!this.targetWindow) throw new Error('Target window is not set');
+        this.connect();
+      }, 200) as unknown as number;
```

Comment:
Can we move this 200 to top of the class as a `const `to follow some clean code principles and give it describtion that is `const intervalConnectionTime = 200` ?

Reply:
I don't think this will add any value to the code.
- We are not going to change the value in the runtime
- value used just once
- value isn’t needed to have an understanding of what the class is doing